### PR TITLE
Create spelling_action.yml

### PR DESCRIPTION
**Pull request by wildsm**

### CODE REVIEW

**Comment by wildsm** [comment]
@mosesyhc - check this out. 

I think we definitely want to delete the duplicate but outdated sdk file in docs/
(which I found through this action)

A separate question is whether we really want this action on every push (instead it could just be an optional, manually deployed action).

File: .github/workflows/spelling_action.yml
Comment (mosesyhc) [inline .github/workflows/spelling_action.yml:1]: Propose to only run spellcheck on `main`, as we seem to only update main for major updates, and that is when spellcheck is certainly required.

**Review by mosesyhc — APPROVED** [review]
Remove duplicated file is good.

See comment in spelling_action.yml for frequency of spellcheck.  We can also switch on manual trigger for this action as well.

File: .github/workflows/spelling_action.yml
Comment (wildsm) [reply .github/workflows/spelling_action.yml:1]: I think this is reasonable; can always fall back to manual

**Pull request merged**